### PR TITLE
add rewards methods for claim generation

### PR DESCRIPTION
**Pull request by shrimalmadhur**
Fixes # .

### What Changed?
- Add Distribution Length Method
- Add Process claim method

### Reviewer Checklist
- [ ] Code is well-documented
- [ ] Code adheres to Go [naming conventions](https://go.dev/doc/effective_go#names)
- [ ] Code deprecates any old functionality before removing it

### CODE REVIEW

File: chainio/clients/elcontracts/writer.go [hunk 1]
```diff
@@ -329,3 +337,29 @@ func (w *ELChainWriter) SetClaimerFor(
...
+	claim rewardscoordinator.IRewardsCoordinatorRewardsMerkleClaim,
+	earnerAddress gethcommon.Address,
+) (*gethtypes.Receipt, error) {
+	if w.rewardsCoordinator == nil {
+		return nil, errors.New("RewardsCoordinator contract not provided")
+	}
+
+	noSendTxOpts, err := w.txMgr.GetNoSendTxOpts()
+	if err != nil {
+		return nil, err
```
Comment (samlaf) [inline chainio/clients/elcontracts/writer.go]: wrap error

File: chainio/clients/elcontracts/writer.go [hunk 2]
```diff
@@ -329,3 +337,29 @@ func (w *ELChainWriter) SetClaimerFor(
...
+	}
+
+	tx, err := w.rewardsCoordinator.ProcessClaim(noSendTxOpts, claim, earnerAddress)
+	if err != nil {
+		return nil, err
```
Comment (samlaf) [inline chainio/clients/elcontracts/writer.go]: wrap error

**Review by samlaf — DISMISSED** [review]

File: chainio/clients/elcontracts/writer.go (hunk 1 above)
Comment (shrimalmadhur) [reply chainio/clients/elcontracts/writer.go]: 9ae2d8729b07949d8007acf29fc240b45791ea6e

File: chainio/clients/elcontracts/writer.go (hunk 2 above)
Comment (shrimalmadhur) [reply chainio/clients/elcontracts/writer.go]: 9ae2d8729b07949d8007acf29fc240b45791ea6e

**Review by samlaf — APPROVED** [review]

**Pull request merged**